lib/rubocop/cop/rails/dynamic_find_by.rb

```diff
@@ -57,6 +61,33 @@ def autocorrect(node)
 
         private
 
+        def allowed?(node)
+          allowed_method?(node) || allowed_receiver?(node) ||
+            whitelisted?(node)
+        end
+
+        def allowed_method?(node)
+          return unless cop_config['AllowedMethods']
+
+          cop_config['AllowedMethods'].include?(node.method_name.to_s)
+        end
+
+        def allowed_receiver?(node)
+          return unless cop_config['AllowedReceivers'] && node.receiver
+
+          cop_config['AllowedReceivers'].include?(node.receiver.source.to_s)
+        end
+
+        # config option `WhiteList` will be deprecated soon
+        def whitelisted?(node)
+          whitelist_config = cop_config['Whitelist']
+          return unless whitelist_config
+
+          whitelist_config.include?(node.method_name.to_s) ||
+            node.receiver &&
+              whitelist_config.include?(node.receiver.source.to_s)
```

Comment:
`Whitelist` should not allow receivers for API compatibility. New API `AllowedReceivers` should be used to add a receiver to the allowed receivers.

```suggestion
          whitelist_config.include?(node.method_name.to_s)
```